backend/models/distance.py

```diff
@@ -220,6 +222,73 @@ def get_matrix(self, source_points, destination_points):
         content = response.json()
         return content['durations']
 
+    def measure_one_to_many_points(self, origin, point_list):
+        """
+        Measure duration from one to mamy points using the matrix API.
+
+        Result is a list of duration in  minutes.
+        """
+        measurements = []
+
+        for batch in iterators.iterate_in_slices(
+            iterable=iter(point_list),
+                batch_size=24):
+            measurements.extend(
+                self.get_matrix(
+                    source_points=[origin],
+                    destination_points=batch)[0][1:]
+            )
+        return measurements
+
+    def closest_with_early_exit(self, origin, point_list, exit_distance):
+        """
+        Find closest point in a list of points, exiting early if exit_distance is reached.
+
+        The exit_distance uses haversine distance (in meters).
+        """
+        haversine_distances = [
+            self._haversine_measurer.measure_between_two_points(origin, point)
+            for point in point_list
+        ]
+        min_haversine_idx, min_haversine_distance = min(
+            enumerate(haversine_distances),
+            key=operator.itemgetter(1)
+        )
+
+        # TODO - Avoid early 'return' so that we can implement one go to
+        # `closest_with_early_exit` function building on `haversine` and `closest`.
+        if min_haversine_distance <= exit_distance:
+            return (
+                min_haversine_distance / TIME_TO_DISTANCE * 60.0, point_list[min_haversine_idx]
+            )
+
+        relevant_points = [
+            point_list[idx]
+            for idx, distance in enumerate(haversine_distances)
+            if distance < min(self.early_exit_outer_radius, 1.5 * min_haversine_distance)
+        ]
+
+        if not relevant_points:
+            return 10**42, point_list[0]
+
+        return self.closest(origin=origin, point_list=relevant_points)
+
+    def closest(self, origin, point_list):
```

Comment:
This function could call `closest_with_early_exit` with `float('inf')` as the exit distance :)

Reply:
Except that it uses closest_with_early_exit  uses closest ;)

Reply:
TRUE.